vendor/templates/vendor/manage/products.html

```diff
@@ -1,50 +1,185 @@
+        <div class="input-group-append">
+          <button class="btn bg-transparent close" type="button" role="close" aria-role="close" id="search_close" style="margin-left: -1.5rem; z-index: 10;">
+            <i class="fa fa-times"></i>
+          </button>
+        </div>
+        <div class="input-group-append">
+          <button id="search_button" class="btn btn-success" style="z-index: 15;">
+            <i class="fas fa-search"></i>
+          </button>
         </div>
       </div>
-      <p><a href="{% url 'vendor_admin:manager-product-create' %}" class="btn btn-primary">{% trans 'Add Product' %}</a></p>
+      <form id="profile_pagination_form" method="get" class="form form-inline text-right">
+        <span class="text-dark mr-2">
+          {% trans 'Show:' %}
+        </span>
+        <select class="selectpicker w-auto" name="paginate_by" id="paginate_by" data-show-tick="true">
+          <option {% if request.GET.paginate_by == '25' %}selected{% endif %}>25</option>
+          <option {% if request.GET.paginate_by == '50' %}selected{% endif %}>50</option>
+          <option {% if request.GET.paginate_by == '100' %}selected{% endif %}>100</option>
+        </select>
+      </form>
+    </div>
+    <div class="table-container table-responsive tc-table">
+      <table class='table table-striped'>
+        <thead>
+          <tr class="d-flex border-0">
+            <th id="product_order" class="text-left col-3{% if '-name' in request.GET.ordering %} desc{% elif 'name' in request.GET.ordering %} asc{% endif %}">{% trans 'Product License' %}</th>
+            <th class="col-2 text-center">{% trans 'Product Offers' %}</th>
+            <th id="availability_order" class="col-2 text-center {% if '-available' in request.GET.ordering %} desc{% elif 'available' in request.GET.ordering %} asc{% endif %}">{% trans 'Availability' %}</th>
+            <th class="ml-auto flex-grow-1"></th>
+          </tr>
+        </thead>
+        <tbody>
+          {% for product in object_list %}
+          <tr class="d-flex">
+            <td class="col-3 text-left">{{product.name}}</td>
+            <td class="col-2 text-center">{{product.offers.count}}</td>
+            <td class="col-2 text-center">{{product.available}}</td>
+            <td class="ml-auto flex-grow-1 text-right">
+              <a class="btn btn-sm btn-primary"
+                href="{% url 'vendor_admin:manager-product-update' product.uuid %}">{% trans 'Edit' %}</a>
+            </td>
+          </tr>
+
+          {% empty %}
+          <tr>
+            <td>{% trans 'No Products' %}</td>
+            <td></td>
+            <td></td>
+            <td></td>
+          </tr>
+          {% endfor %}
+        </tbody>
+      </table>
+    </div>
+
+
+    <div class="pagination justify-content-between">
+      <span>
+        {% blocktrans with object_list|length as total_entries %}
+        Showing {{page_obj.number}} to {{page_obj.number + page_obj.paginate_by }} of {{total_entries}}
```

Comment:
you should be able to use page_obj.start_index and end_index for this part

Reply:
changed thanks